server/ingest/http/handler.go

```diff
@@ -96,42 +97,16 @@ func (h *httpHandler) removeProvider(w http.ResponseWriter, r *http.Request) {
 }
 
 // ----- ingest handlers -----
-// PUT /ingest/advertisement
-func (h *httpHandler) advertise(w http.ResponseWriter, r *http.Request) {
-	/*
-		w.Header().Set("Content-Type", "application/json")
-		adBuild := ingestion.Type.Advertisement.NewBuilder()
-		err := dagjson.Decoder(ptp, r.Body)
-		if err != nil {
-			log.Errorw("Advertise request json decode", "err", err.Error())
-			http.Error(w, "", http.StatusBadRequest)
-			return
-		}
-
-		_, err := h.registry.Advertise(r.Context(), adBuild.Build().(ingestion.Advertisement))
-		if err != nil {
-			log.Errorw("Advertise failed:", "err", err)
-			http.Error(w, "", http.StatusInternalServerError)
-			return
-		}
-	*/
-	w.WriteHeader(http.StatusNoContent)
-}
+// PUT /ingest/announce
+func (h *httpHandler) announce(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
 
-// POST /ingest/content
-func (h *httpHandler) indexContent(w http.ResponseWriter, r *http.Request) {
-	body, err := io.ReadAll(r.Body)
+	defer r.Body.Close()
+	err := h.ingestHandler.Announce(r.Context(), r.Body)
 	if err != nil {
-		log.Errorw("failed reading body", "err", err)
+		log.Errorw("Advertise failed:", "err", err)
 		http.Error(w, "", http.StatusInternalServerError)
```

Comment:
```suggestion
		httpserver.HandleError(w, err, "announce") 
```
This will handle the correct HTTP status for the error.